core/ast_node.js

```diff
@@ -408,56 +448,62 @@ Blockly.ASTNode.prototype.findTopASTConnection_ = function(block) {
 /**
  * Given a location in a stack of blocks find the next out connection. If the
  * location is nested the next out location should be the connected input.
- * @param {!Blockly.Block} block The source block for the current location.
- * @return {?Blockly.ASTNode} The ast node holding thenext out connection or
+ * @param {Blockly.Block} block The source block for the current location.
+ * @return {Blockly.ASTNode} The ast node holding thenext out connection or
```

Comment:
Pattern: when your function returns null in some cases, the @return description can contain "or null if <condition>".  Then someone who isn't carefully parsing the types of the comments still knows that it might return null and has some information about why.